python/ray/serve/tests/test_cli.py

```diff
@@ -1417,5 +1417,39 @@ def test_run_with_auto_address(
         assert expected_warning_message not in logs
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
+def test_run_config_request_timeout(ray_start_stop):
+    """Test running serve with request timeout http_options.
+
+    When serve run with config file having 0.1s as the `request_timeout_s` in
+    the `http_options`. First case tests when the deployment runs longer than the 0.1s
+    timeout and returned task failed message. The second case is testing the deployment
+    runs shorter than the 0.1s timeout and it should return success message.
+    """
+    config_file_name = os.path.join(
+        os.path.dirname(__file__),
+        "test_config_files",
+        "http_option_request_timeout_s.yaml",
+    )
+    p = subprocess.Popen(["serve", "run", config_file_name])
+
+    # Ensure the http request is killed and failed when the deployment runs longer than
+    # the 0.1 request_timeout_s set up in the config yaml
+    wait_for_condition(
+        lambda: requests.get("http://localhost:8000/app1?sleep_s=0.11").text
+        == "Task failed with 10 retries.",
```

Comment:
Failing with 10 retries is going to take a while because we use an exponential backoff. Is there anything we can do here to speed things up?

Maybe we could deploy this using the Python API and then use a `SignalActor` to let the request succeed on the second retry?

Reply:
Agreed, let me see how to stub this out